yarn.lock

```diff
@@ -12686,13 +12659,27 @@ minimalistic-crypto-utils@^1.0.1:
   resolved "https://registry.yarnpkg.com/minimalistic-crypto-utils/-/minimalistic-crypto-utils-1.0.1.tgz#f6c00c1c0b082246e5c4d99dfb8c7c083b2b582a"
   integrity sha1-9sAMHAsIIkblxNmd+4x8CDsrWCo=
 
-"minimatch@2 || 3", minimatch@5.0.1, minimatch@^3.0.4, minimatch@^3.0.5, minimatch@^3.1.2, minimatch@~3.0.4:
+"minimatch@2 || 3", minimatch@^3.0.4, minimatch@^3.1.2:
   version "3.1.2"
   resolved "https://registry.yarnpkg.com/minimatch/-/minimatch-3.1.2.tgz#19cd194bfd3e428f049a70817c038d89ab4be35b"
   integrity sha512-J7p63hRiAjw1NDEww1W7i37+ByIrOWO5XQQAzZ3VOcL0PNybwpfmV/N05zFAzwQ9USyEcX6t3UO+K5aqBQOIHw==
   dependencies:
     brace-expansion "^1.1.7"
 
+minimatch@5.0.1:
+  version "5.0.1"
+  resolved "https://registry.yarnpkg.com/minimatch/-/minimatch-5.0.1.tgz#fb9022f7528125187c92bd9e9b6366be1cf3415b"
+  integrity sha512-nLDxIFRyhDblz3qMuq+SoRZED4+miJ/G+tdDrjkkkRnjAsBexeGpgjLEQ0blJy7rHhR2b93rhQY4SvyWu9v03g==
+  dependencies:
+    brace-expansion "^2.0.1"
+
+minimatch@~3.0.4:
```

Comment:
Do we want to push this to `3.1.2` like before?

Reply:
I don't think we need to add a manual resolution in this case since `3.0.8` addresses the CVE that necessitated the bump to `3.0.5` previously. That could have been addressed without a manual resolution by bumping the patch version in the yarn.lock file (remove the version and then re-bootstrap).

Reply:
@tmarkley @miki, not related to this specific PR, but are there reference docs to understand what these changes are and how we need to apply them w.r.t OSD?  Since the yarn lock file is an auto generated file and usually automatically managed, I am not too familiar as to why the inital manual resolution was necessary and why we are removing it now. I understand your reasoning for the change @tmarkley but i do not know enough to know:
1. Why we need to manage this manually
2. Why we do it the way we do it today.

Having a common understanding of its workings will help others who have to upgrade the dependencies correctly too.

Reply:
@ashwin-pc yes, I'm going open a PR to include that kind of information in the SECURITY file for managing CVEs and dependencies. If you look at the commit history/git blame, you can see that the resolution was added in #2640.

Overall, it is correct that the yarn.lock file should not be modified manually, but adding [selective dependency resolutions](https://classic.yarnpkg.com/en/docs/selective-version-resolutions) can cause issues with certain libraries due to compatibility issues (e.g. if a library in the package.json depends on an older version of an upstream library that was resolved to a newer version that breaks previous functionality). I think it's best to strive to avoid resolutions whenever possible and only reserve them for situations where we're resolving a CVE and have a longer-term plan to upgrade/remove/change dependencies and then remove the resolution once that work is done.